core/market/src/protocol/discovery.rs

```diff
@@ -94,18 +140,57 @@ impl Discovery {
         &self,
         offer_ids: Vec<SubscriptionId>,
     ) -> Result<(), DiscoveryError> {
+        log::trace!("bcast_unsubscribes {}", offer_ids.len());
         if offer_ids.is_empty() {
             return Ok(());
         }
-        let default_id = self.default_identity().await?;
+        let must_schedule = {
+            let mut queue = self.inner.unsub_queue.lock().await;
+            let result = queue.len() == 0;
+
+            queue.append(&mut offer_ids.clone());
+            result
+        };
+        log::trace!(
+            "bcast_unsubscribes done appending. must_schedule? {}",
+            must_schedule
+        );
+
+        if must_schedule {
+            let myself = self.clone();
+            let _ = Arbiter::spawn(async move {
```

Comment:
Is it enough to just spawn without delay?

Reply:
that would defeat the whole point, the bcast's are send as they are called

Reply:
No. I meant spawn closure with delay (very short) inside to wait for some other subscribes. Because I don't understand why it will batch broadcasts. They must be really close and we need lack with scheduling

Reply:
you are correct, the delay was missing and i was not aware of that.

Thanks for pointing it out! great catch :)

Fixed :D